CONTRIBUTING.md

```diff
@@ -0,0 +1,170 @@
+# Welcome to Fosscord Contributing Guide
+
+
+This document aims to get you started with contributing to this repo! 
+
+Please note that currently our Rendered documentation is out of date. It is accurate for the master branch as of writing but its not always accurate for Staging.
+
+Please see the [contributors' guide](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html) in our rendered documentation.
+
+# 1. Who can contribute to Fosscord?
+
+Everyone is welcome to contribute code to [Fosscord GitHub Organisation Projects](https://github.com/fosscord), 
+provided that they are willing to license their contributions under the 
+same license as the project itself. We follow a simple 'inbound=outbound' model 
+for contributions: the act of submitting an 'inbound' contribution means that the 
+contributor agrees to license the code under the same terms as the project's overall 'outbound'
+license - in our case, this is almost always GNU Affero General Public License v3.0 (see
+[LICENSE](https://github.com/fosscord/fosscord-server/blob/master/COPYING)).
+
+# 2. What do I need?
+
+Fosscord Server is written in TypeScript and requires [a NodeJS version of 14 or above](https://nodejs.org/en/download/).
+
+The source code for Fosscord Server is hosted on GitHub. You will also need a Git client.
+
+To build Fosscord Server you will need Python 3 due to dependencies needing it. A currently supported version of Python 3 is required 
+as issues that cant be replicated using non EOL Python versions will be closed.
+
+
+# 3. Get the source.
+
+The preferred and easiest way to contribute changes is to fork the relevant
+project on GitHub, and then [create a pull request](https://help.github.com/articles/using-pull-requests/) 
+to ask us to pull your changes into our repo.
+
+Please base your changes on the `staging` branch.
+
+```sh
+git clone git@github.com:YOUR_GITHUB_USER_NAME/fosscord-server.git
+git checkout staging
+```
+
+If you need help getting started with git, this is beyond the scope of the document, but you
+can find many good git tutorials on the web.
+
+# 4. Install the OS Dependent Dependencies
+
+## Under Linux
+
+You need to install `gcc` and `g++`. Packaged with `build-essential` on Debian/Ubuntu and `base-devel` on Arch
+
+
+## Under Windows
+
+[Visual Studio](https://visualstudio.microsoft.com/) with the C++ package. 
+
+# 5. Get in touch.
+
+Join our developer community on Discord: https://discord.gg/zS5sjnCkSG
+
+# 8. Test, test, test!
+<a name="test-test-test"></a>
+
+While you're developing and before submitting a patch, you'll
+want to test your code.
+
+## Create Database Migrations for your code.
+
+CHANGE ME

```

Comment:
Ask Arcane or myself about this one.
We both suck at writing docs, otherwise we'd write it ourselves :P

Reply:
`node scripts/db_migrations.js <optionally, name of your migration, though itll prompt>` for migrations.
Would recommend setting FC_DB_POSTGRES and FC_DB_MARIADB in system-wide environment variables (~/.profile is good enough on linux), these hold the default values for connection strings for those DBMSses for migrations.

Reply:
@ChrisChrome would you like my help in writing our developer documentation? I can describe things elaborately.

Reply:
I think you should change from `staging` branch to `master` branch.